lib/uring/include/discover.ml

```diff
@@ -2,49 +2,122 @@ module C = Configurator.V1
-          "POLLERR", Int;
-          "POLLHUP", Int;
+      let defs =
+        C.C_define.import c ~c_flags:["-D_GNU_SOURCE"; "-I"; Filename.concat (Sys.getcwd ()) "include"]
+          ~includes:["fcntl.h"; "poll.h"; "sys/uio.h"; "linux/time_types.h"; "liburing.h"]
+          C.C_define.Type.[
+            "POLLIN", Int;
+            "POLLOUT", Int;
+            "POLLERR", Int;
+            "POLLHUP", Int;
 
-          "O_RDONLY", Int;
-          "O_WRONLY", Int;
-          "O_RDWR", Int;
-          "O_CREAT", Int;
-          "O_EXCL", Int;
-          "O_NOCTTY", Int;
-          "O_TRUNC", Int;
-          "O_APPEND", Int;
-          "O_NONBLOCK", Int;
-          "O_DSYNC", Int;
-          "O_DIRECT", Int;
-          "O_LARGEFILE", Int;
-          "O_DIRECTORY", Int;
-          "O_NOFOLLOW", Int;
-          "O_NOATIME", Int;
-          "O_CLOEXEC", Int;
-          "O_SYNC", Int;
-          "O_PATH", Int;
-          "O_TMPFILE", Int;
+            "O_RDONLY", Int;
+            "O_WRONLY", Int;
+            "O_RDWR", Int;
+            "O_CREAT", Int;
+            "O_EXCL", Int;
+            "O_NOCTTY", Int;
+            "O_TRUNC", Int;
+            "O_APPEND", Int;
+            "O_NONBLOCK", Int;
+            "O_DSYNC", Int;
+            "O_DIRECT", Int;
+            "O_LARGEFILE", Int;
+            "O_DIRECTORY", Int;
+            "O_NOFOLLOW", Int;
+            "O_NOATIME", Int;
+            "O_CLOEXEC", Int;
+            "O_SYNC", Int;
+            "O_PATH", Int;
+            "O_TMPFILE", Int;
 
-          "AT_FDCWD", Int;
+            "AT_FDCWD", Int;
 
-          "sizeof(struct iovec)", Int;
-          "sizeof(struct __kernel_timespec)", Int;
-        ]
-      |> List.map (function
-          | name, C.C_define.Value.Int v ->
-            let name = 
-              match name with
-              | "sizeof(struct iovec)" -> "sizeof_iovec"
-              | "sizeof(struct __kernel_timespec)" -> "sizeof_kernel_timespec"
-              | nm -> nm 
-            in
-            Printf.sprintf "let %s = 0x%x" (String.lowercase_ascii name) v
-          | _ -> assert false
-        )
-      |> C.Flags.write_lines "config.ml"
+            "sizeof(struct iovec)", Int;
+            "sizeof(struct __kernel_timespec)", Int;
+          ]
+            |> List.map (function
+                | name, C.C_define.Value.Int v ->
+                  let name = 
+                    match name with
+                    | "sizeof(struct iovec)" -> "sizeof_iovec"
+                    | "sizeof(struct __kernel_timespec)" -> "sizeof_kernel_timespec"
+                    | nm -> nm 
+                  in
+                  Printf.sprintf "let %s = 0x%x" (String.lowercase_ascii name) v
+                | _ -> assert false
+          )
+      in
+      let ops =
+        C.C_define.import c ~c_flags:["-D_GNU_SOURCE"; "-I"; Filename.concat (Sys.getcwd ()) "include"]
+          ~includes:["fcntl.h"; "poll.h"; "sys/uio.h"; "linux/time_types.h"; "liburing.h"]
```

Comment:
Same as above comment. `linux/time_types.h` is not needed if `liburing.h` can be included. Unless of course some other symbols defined in `linux/time_types.h` is needed.

Reply:
I've removed everything except `liburing.h` here.